### ISSUE TITLE
Create TOS & Country screen for Android 

### ISSUE THREAD

**Issue by Odrakir** [toggl/mobileapp#7002]
After the authorisation from Google (or Apple) or after filling email/password when signing up, the user should be presented with a checkbox to accept the terms of service and a way to select his country.

This will be a modal dialog presented from either the main onboarding screen or the email signup screen. It needs to return the country and wether the user accepted or not the TOS to the presenting ViewModel.

#7000 should be done first.

Designs and full specs: https://www.notion.so/toggl/Designs-wip-d767ebc157c542c790e1bc93eeaf006f#835299ea40344044b2572f356b4f47c1

Android designs are pending, but we can start implementation taking iOS designs as reference and then do the appropriate changes later.

**NOTE:** This issue is part of the `Onboarding refactor` project and it should be merged into the project branch and not into develop.

**Comment by daividssilverio** [comment]
Closed via #7138
